### PR TITLE
Remove redundant suggestions for un-invoked macros

### LINKED ISSUE #67740: Redundant `use` suggestion for incorrectly invoked macro
The `use` suggestion shouldn't be emitted as `format` is already in scope.

```
error[E0423]: expected function, found macro `format`
   --> src/librustc_typeck/check/pat.rs:627:29
    |
627 |                     e.note(&format("`{}` is interpreted as a constant pattern, not a new blinding", qpath));
    |                             ^^^^^^
    |
help: use `!` to invoke the macro
    |
627 |                     e.note(&format!("`{}` is interpreted as a constant pattern, not a new blinding", qpath));
    |                             ^^^^^^^
help: possible better candidate is found in another module, you can import it into scope
    |
1   | use std::fmt::format;
    |
```

_Edit:_ The suggestion is _correct_: `std::fmt::format` is _not_ the `format!()` macro. That being said, the import suggestion should have slightly different wording if other suggestions are provided. Changing it to "alternatively, another candidate is found..." would make it clearer, but this indeed _not_ a bug.

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #67740

r? @estebank

### CODE REVIEW

File: src/librustc_resolve/late/diagnostics.rs [hunk 1]
```diff
@@ -282,15 +282,19 @@ impl<'a> LateResolutionVisitor<'a, '_> {
...
                 res,
                 &path_str,
                 &fallback_label,
             ) {
-                return (err, candidates);
+                if needs_candidates {
+                    return (err, candidates);
+                } else {
+                    return (err, Vec::new());
+                }
```
Comment (BO41) [inline src/librustc_resolve/late/diagnostics.rs]: Maybe "optimize" to
```rust
                return if needs_candidates {
                    (err, candidates)
                } else {
                    (err, Vec::new())
                };
```

Comment (estebank) [reply src/librustc_resolve/late/diagnostics.rs]:
```
return (err, if needs_candidates {
    candidates,
} else {
   vec![]
});
```
or similar would work too :)

**Comment by estebank** [comment]
r=me after resolving the nitpick

**Comment by petrochenkov** [comment]
https://github.com/rust-lang/rust/issues/67740 is not a bug, it's an entirely legitimate suggestion.
Even if *macro* `format` is in scope, we still can recommend the *function* candidate [`std::fmt::format`](https://doc.rust-lang.org/nightly/std/fmt/fn.format.html) because the code in the example actually calls a function and not a macro.

If there are two possible mistakes (missing `!` or forgotten function import), then we are not trying to guess which one was made and show suggestions for both.
So, I don't think this PR is an improvement.

**Comment by estebank** [comment]
I didn't realize that it was suggesting a function, not the macro itself. Because of that the suggestion is correct but still mildly confusing. We might want to tweak the wording of the suggestion if there's more than one and nothing else instead.

**Comment by JohnTitor** [comment]
So, we should close this PR and continue the discussion on that issue?

**Comment by petrochenkov** [comment]
@estebank 
>We might want to tweak the wording of the suggestion if there's more than one and nothing else instead.

Right now we say "possible better candidate" instead of "possible candidate" when we see an unsuitable candidate in scope (the macro) and a suitable candidate elsewhere (the function), but perhaps some better wording can be made.

@JohnTitor 
>So, we should close this PR and continue the discussion on that issue?

Yeah, I'll close this PR, sorry for your time.

**Pull request closed without merging**